lua/lspconfig/server_configurations/sonarlint.lua

```diff
@@ -0,0 +1,92 @@
+local util = require 'lspconfig.util'
+
+local tcp = vim.loop.new_tcp()
+tcp:bind('127.0.0.1', 0)
+local port = tostring(tcp:getsockname().port)
+
+local root_files = {
+  'compile_commands.json',
+  '.ccls',
+
+  'package.json',
+  'tsconfig.json',
+
+  'composer.json',
+
+  '.git',
+}
+
+return {
+  default_config = {
+    cmd = { 'nc', '-l', port },
+    filetypes = { 'c', 'cpp', 'html', 'javascript', 'php', 'python', 'typescript', 'xml' },
+    root_dir = function(file, bufnr)
+      if (vim.fn.getbufvar(bufnr, '&filetype')) ~= 'python' then
```

Comment:
there can be `vim.bo.filetype`. and why there judge the filetype ~= python your filetypes not include python

Reply:
@glepnir There is a bug with sonarlint python analyzer that slows the sonarlint language server down a lot if `root_dir` or `workspace_folders` is configured. That's why I excluded python files from `root_dir` configuration.